src/data/matrix/hadamard.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.notation
+import linear_algebra.matrix.trace
+import data.complex.basic
+
+/-!
+# Hadamard product of matrices
+
+This file defines the Hadamard product `matrix.Hadamard`
+and contains basic properties about them.
+
+## Main definition
+
+- `matrix.Hadamard`: defines the Hadamard product,
+  which is the pointwise product of two matrices of the same size.
+
+## Notation
+
+* `⊙`: the Hadamard product `matrix.Hadamard`;
+
+## References
+
+*  <https://en.wikipedia.org/wiki/Hadamard_product_(matrices)>
+
+## Tags
+
+Hadamard product, Kronecker product, Hadamard, Kronecker
+-/
+
+variables {α β γ I J K L M N: Type*}
+variables {R : Type*}
+variables {m n p q r s t: ℕ}
+variables [fintype I] [fintype J] [fintype K] [fintype L] [fintype M] [fintype N]
+
+namespace matrix
+open_locale matrix big_operators
+open complex
+
+def Hadamard [has_mul α] (A : matrix I J α) (B : matrix I J α) :
```

Comment:
I think the uppercase may be better than the lowercase as it is a name. Is it necessary to match "kronecker"?

Reply:
Personally, I prefer to change "kronecker" to "Kronecker"

Reply:
I agree they should be consistent; I'd recommend asking Zulip to decide.

Reply:
If both are OK, I can change "kronecker" to "Kronecker" in a PR I will need to that file.

Reply:
I think currently capitalised names are used in the category theory library pretty much exclusively; this will change in Lean4 though

Reply:
@ericrbg Do you mean all the names will be changed to be in lowercase in category theory? I did not quite get what you mean by "this".

Reply:
What @ericrbg is referring to is that the naming conventions used in Lean 3 is `snake_case`, whereas in Lean 4, it has been changed to `PascalCase`. Currently in mathlib we use `snake_case` for everything except categories, which are `PascalCase`. In Lean 4 we will have to change things.

Reply:
I think it should be lower case for this PR, across all of mathlib names in identifiers are lowercase, e.g. cauchy, urysohns etc. If the community does want to change this that's a later job for a separate PR as it will be a huge change.